### ISSUE TITLE
Concatenating ordered strings produces a string which (incorrectly) thinks it is sorted

### ISSUE THREAD

**Issue by ChrisJefferson** [gap-system/gap#733]
Wandered across this while testing and fixing #730. The last value is wrong, c isn't sorted. This is at least as old as v 4.7.2

```
gap> a := "abc";
"abc"
gap> b := "def";
"def"
gap> IsSortedList(a);
true
gap> IsSortedList(b);
true
gap> c := Concatenation(b,a);
"defabc"
gap> HasIsSortedList(c);
true
gap> IsSortedList(c);
true
```

**Comment by ChrisJefferson** [comment]
Deeper: This is 'Append' on strings

**Comment by markuspf** [comment]
but if you do the same thing that `Append` does on the REPL the error doesn't occur...

**Comment by ChrisJefferson** [comment]
Found it, looking for other issues

**Comment by markuspf** [comment]
I have to admit that I am a little bit confused as to why `FuncAPPEND_LIST_INTR` is selected/called there?

**Comment by ChrisJefferson** [comment]
Looks like `ApplicableMethod` doesn't understand `DeclareOperationKernel`.

Basically, Append calls APPEND_LIST, which is FuncAPPEND_LIST in listfunc.c, which special cases built-in types, and only then dispatches.

Actually, it's not obvious how ApplicableMethod could know when the C code will intercept. It could at least warn that it doesn't know!

**Comment by markuspf** [comment]
I see. I'll keep a note for the type exploration tools, and keep a mental note to chase up kernel operations when doing this kind of debugging.

**Comment by markuspf** [comment]
This is fixed with the merge of #735
